core/src/main/scala/shapeless/ops/hlists.scala

```diff
@@ -946,12 +946,22 @@ object hlist {
 
     type Aux[L <: HList, A <: Nat, B <: Nat, Out0 <: HList] = SelectRange[L, A, B] {type Out = Out0}
 
-    implicit def SelectRangeAux[L <: HList, A <: Nat, B <: Nat, Ids <: HList, SelOut <: HList]
-    (implicit range: shapeless.ops.nat.Range.Aux[A, B, Ids], sel: SelectMany.Aux[L, Ids, SelOut]): Aux[L, A, B, SelOut] =
-      new SelectRange[L, A, B] {
-        type Out = SelOut
+    implicit def thenTake[L <: HList, T <: Nat](
```

Comment:
Not sure about the name of this method. Why not just call it `take`?

Reply:
Then i should rename also a param name. Ok, i have an idea.